src/components/Board.vue

```diff
@@ -1,89 +1,129 @@
+      bottomGoodArea: goodArea.getBoundingClientRect().bottom,
+      topExcellentArea: excellentArea.getBoundingClientRect().top,
+      bottomExcellentArea: excellentArea.getBoundingClientRect().bottom
     }
-  this.scoreElement!.textContent = this.scoreCounter.toString()
   }
 
-  delay = (ms: number) => {
-    return new Promise(resolve => setTimeout(resolve, ms))
+  startGame (): void {
+    console.log('the game is on')
+    this.isActive = true
+    this.runRender()
+    window.addEventListener('keydown', this.onKeyDown)
   }
 
-  renderBlocks = async (elements: any) => { // function will be deleted
-    for (let i = 0; i < elements.length; i += 1) {
-      if (i === 0) {
-        this.gameFlow?.appendChild(elements[i]) // components will be added using VUE methods
+  runRender (): void {
+    const interval = setInterval(() => {
+      this.addArrow()
+      this.safeLoop += 1
+      clearInterval(interval)
+      if (this.safeLoop < 1000 && this.isActive) {
+        this.runRender()
       } else {
-        await this.delay(this.block.getRandom().timeout)
-        this.gameFlow?.appendChild(elements[i])
+        this.isActive = false
+        console.log('stop')
+        setTimeout(() =>
+          window.removeEventListener('keydown', this.onKeyDown),
+        5000) // as an animation duration, to keep listener exist till arrows exist
       }
+    }, this.getRandom(200, 2000))
+  }
 
-      elements[i].addEventListener('animationend', () => {
-        document.getElementById(elements[i].id)?.remove() // components will be deleted using VUE methods
-      })
-    }
+  stopGame (): void {
+    this.isActive = false
   }
 
-  startGame () {
-    this.counts += 1
-    console.log('the game is on')
-    const elements = []
-    for (let i = 1; i <= this.block.getRandom().elements; i += 1) {
-      elements.push(new Block().createItem())
+  onKeyDown (e: KeyboardEvent): void {
+    const key: string = e.key
+    const keyMap: { [index: string]: EDirection } = {
+      ArrowLeft: EDirection.ArrowLeft,
+      ArrowUp: EDirection.ArrowUp,
+      ArrowDown: EDirection.ArrowDown,
+      ArrowRight: EDirection.ArrowRight
+    }
+    const firstFondArrow = this.$store.state.arrowStore.arrowsData.find((e: IArrowData) =>
+      e.direction === keyMap[key]
+    )
+    if (firstFondArrow) {
+      this.checkArrowPosition(firstFondArrow.id)
     }
-    this.renderBlocks(elements)
-    document.addEventListener('keydown', this.logKey)
   }
 
-  logKey (e: KeyboardEvent) {
-    const key: any = e.key
-    const keyToColumn: any = {
-      ArrowLeft: document.getElementsByClassName('left-arrow'),
-      ArrowUp: document.getElementsByClassName('up-arrow'),
-      ArrowDown: document.getElementsByClassName('down-arrow'),
-      ArrowRight: document.getElementsByClassName('right-arrow')
+  checkArrowPosition (id: number): void {
+    const arrowEl: HTMLElement = document.getElementById(id.toString())!
```

Comment:
do you check if `getElementById` returns `undefined`? `!` isn't a check

Reply:
thanks, will fixed it in next commit